extensions/src/main/java/org/pentaho/platform/web/http/api/resources/RepositoryResource.java

```diff
@@ -889,11 +893,15 @@ protected Response getContentGeneratorResponse( CGFactory fac ) {
       rsc( "Nope, [{0}] is not a content generator ID.", fac.getContentGeneratorId() ); //$NON-NLS-1$
       return null;
     }
-    rsc(
-        "Yep, [{0}] is a content generator ID. Executing (where command path is {1})..", fac.getContentGeneratorId(),
-        fac.getCommand() ); //$NON-NLS-1$
-    GeneratorStreamingOutput gso = fac.getStreamingOutput( contentGenerator );
-    return Response.ok( gso ).build();
+    Response response = checkPermissionIfUserIsEditingContent(fac.getContentGeneratorId());
```

Comment:
You're right that it is `fac.getContentGeneratorId()` (and not `fac.getCommand()`, as I had suggested in https://github.com/pentaho/pentaho-platform/pull/5131#discussion_r963805183) that matters for our purposes! I had misread the code.